zerver/models.py

```diff
@@ -747,6 +747,13 @@ def webathena_enabled(self) -> bool:
     def presence_disabled(self) -> bool:
         return self.is_zephyr_mirror_realm
 
+    def is_web_public(self) -> bool:
+        """
+        If any of the streams in the realm is web
+        public, then the realm is web public.
+        """
+        return Stream.objects.filter(realm=self, is_web_public=True).exists()
```

Comment:
We generally try to avoid properties that can do database queries, since that's not intuitive behavior.

Let's call this `has_web_public_streams()`, which sounds more like a thing that might do some work.

Also, can you have this check `realm.plan_type != Realm.LIMITED` first?  I think we'll want to limit this feature in that way to make it harder to use Zulip Cloud to host spam using this feature.

Reply:
Done.